MozillaFoundation/foundation.mozilla.org
# Add Custom Privacy Text Option to CTAs

# Description

This PR adds a field to CTAs to overwrite the default privacy notice. This new field will allow users to A/B test custom privacy text vs. the default privacy text on various signups or petitions.

When this field is set, it will display instead of the default privacy notice. When this field is empty (as by default), the default privacy text will display.

Link to sample test page: 
Related PRs/issues: #12169 / TP1-394

Credentials:
username: admin
password: W5YO6CFsu0$98!ch

# To Test
1) Log in to review app via https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/cms/ (see credentials above)
2) Navigate to https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/en/campaigns/century-interview-plan/
3) Note the default privacy text
4) Edit this page
5) Click on the `...` actions to edit the Cta
6) Note the new field for Privacy Notice and the help text
7) Add in a new Privacy Notice, and be sure to include a link within the richtext field by highlighting some text.
8) Review the campaign page again: https://foundation-s-tp1-394-12-7kcnly.herokuapp.com/en/campaigns/century-interview-plan/
9) Note the custom Privacy Notice is now showing

# Checklist



**Tests**
~~- [ ] Is the code I'm adding covered by tests?~~

**Changes in Models:**
~~- [ ] Did I update or add new fake data?~~
~~- [ ] Did I squash my migration?~~

**Documentation:**
~~- [ ] Is my code documented?~~
~~- [ ] Did I update the READMEs or wagtail documentation?~~

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP1-610)
